code/headers/st7735_buffered.hpp

```diff
@@ -30,19 +41,38 @@ namespace r2d2::display {
          * @param y
          * @param data
          */
-        void set_pixel(uint16_t x, uint16_t y, const uint16_t data) override;
+        void set_pixel(uint16_t x, uint16_t y, const uint16_t data) override {
+
+            const uint16_t t =
+                __REV16(data); // make a copy and reverse byte order
+
+            // write pixel data to the buffer
+            this->buffer[x + (y * this->width)] = t;
+        }
 
         /**
-         * @brief Directly write multiple pixels to the screen
+         * @brief Directly write multiple pixels to the screen. This effectively
+         * draws a rectangle based on given location and size.
+         *
          *
          * @param x
          * @param y
          * @param width
          * @param height
-         * @param data
+         * @param data Data is a pointer to the color of the pixel
          */
         void set_pixels(uint16_t x, uint16_t y, uint16_t width, uint16_t height,
-                        const uint16_t *data) override;
+                        const uint16_t *data) override {
+            // unfortunaly the arduino due is little endian otherwise we could
+            // put all the data directly to the buffer
+            for (std::size_t v = 0; v < height; v++) {
```

Comment:
The variable names `v` and `t` make it really hard to reason about what is actually happening in the body of the loop. Could you refactor this?

Reply:
should allready be fixed in the latest version